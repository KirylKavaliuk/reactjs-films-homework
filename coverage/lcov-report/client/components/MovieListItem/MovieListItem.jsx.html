<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for client/components/MovieListItem/MovieListItem.jsx</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="..\..\..\prettify.css" />
    <link rel="stylesheet" href="..\..\..\base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(..\..\..\sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="..\..\..\index.html">All files</a> / <a href="index.html">client/components/MovieListItem</a> MovieListItem.jsx
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>5/5</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>2/2</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>2/2</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>5/5</span>
      </div>
    </div>
  </div>
  <div class='status-line high'></div>
<pre><table class="coverage">
<tr><td class="line-count quiet">1
2
3
4
5
6
7
8
9
10
11
12
13
14
15
16
17
18
19
20
21
22
23
24
25
26
27
28
29
30
31
32
33
34
35
36
37
38
39
40
41
42
43
44
45
46
47
48
49
50
51
52
53
54
55
56
57
58
59</td><td class="line-coverage quiet"><span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">3x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">3x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">3x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">3x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span></td><td class="text"><pre class="prettyprint lang-js">import React from 'react';
import PropTypes from 'prop-types';
&nbsp;
import { withDialogContext } from 'utils/dialog';
import withConditionalRendering from 'utils/rendering';
&nbsp;
import Image from 'components/Image/Image';
import Button from 'components/Button/Button';
import Rating from 'components/Rating/Rating';
import Link from 'components/Link/Link';
&nbsp;
import styles from './MovieListItem.scss';
&nbsp;
const MovieListItem = ({ movie }) =&gt; (
  &lt;div className={ styles.movieItem }&gt;
    &lt;Image db className={ styles.poster } width={ 1280 } src={ movie.backdrop_path }/&gt;
&nbsp;
    &lt;div className={ styles.info }&gt;
      &lt;header className={ styles.header}&gt;
        &lt;h1 className={ styles.title }&gt;{ movie.title }&lt;/h1&gt;
        &lt;p className={ styles.genres }&gt;
          { movie.genres &amp;&amp; movie.genres.slice(0, 3).map((genre, index) =&gt; (
              &lt;Link
                className={ styles.genre }
                key={ index }
                to={ `/genre/${genre.id}` }
                clearParams={ ['query', 'movie'] }
              &gt;
                { genre.name }
              &lt;/Link&gt;
          )) }
          &lt;/p&gt;
        &lt;div className={ styles.rating }&gt;
          &lt;Rating value={ movie.vote_average / 2 }/&gt;
        &lt;/div&gt;
      &lt;/header&gt;
&nbsp;
      &lt;p className={ styles.descriptionText }&gt;
        { movie.overview }
      &lt;/p&gt;
&nbsp;
      &lt;Link params={{ trailer: movie.id }}&gt;
        &lt;Button label='Watch Now' className={ styles.watchButton }
        /&gt;
      &lt;/Link&gt;
    &lt;/div&gt;
  &lt;/div&gt;
);
&nbsp;
MovieListItem.defaultProps = {
  movie: null,
};
&nbsp;
MovieListItem.propTypes = {
  movie: PropTypes.object,
};
&nbsp;
export default withDialogContext(withConditionalRendering(MovieListItem, 'movie'));
&nbsp;</pre></td></tr>
</table></pre>
<div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="https://istanbul.js.org/" target="_blank">istanbul</a> at Fri Feb 01 2019 11:31:09 GMT+0300 (GMT+03:00)
</div>
</div>
<script src="..\..\..\prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="..\..\..\sorter.js"></script>
</body>
</html>
