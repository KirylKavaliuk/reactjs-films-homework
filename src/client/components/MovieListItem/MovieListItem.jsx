import React, { Component } from 'react';
import PropTypes from 'prop-types';

import withConditionalRendering from 'utils/withConditionalRendering';
import { withDialogContext } from 'utils/dialog';

import classNames from 'classnames';

import Image from 'components/Image/Image';
import Icon from 'components/Icon/Icon';
import Button from 'components/Button/Button';
import Video from 'components/Video/Video';
import Rating from 'components/Rating/Rating';

import styles from './MovieListItem.scss';

class MovieListItem extends Component {
  state = {
    descriptionOpen: false,
  };

  toggleDescriptionHandler = () => {
    const { descriptionOpen } = this.state;
    this.setState({ descriptionOpen: !descriptionOpen });
  }

  render() {
    return (
      <div className={ styles.movieItem }>
        <Image
          className={ styles.poster }
          width={ 1280 }
          src={ `db/${this.props.movie.backdrop_path}` }
        />

        { !this.state.descriptionOpen && <div className={ styles.more }>
          <div
            className={ styles.playButton }
            onClick={ () => this.props.openDialog(<Video id={ this.props.movie.id }/>) }
          >
            <Icon
              className={ styles.playButtonIcon }
              name='play'
            />
            <h3 className={ styles.playButtonHeader }>Play Now</h3>
          </div>

          <Button
            label='View Info'
            transparent
            onClick={ this.toggleDescriptionHandler }
          />
        </div>
        }

        <div className={
          classNames(
            styles.info,
            { [styles.infoVisible]: this.state.descriptionOpen },
          )}
        >
          <div className={
              classNames(
                styles.cancelIcon,
                { [styles.cancelIconVisible]: this.state.descriptionOpen },
              )
            }
            onClick={ this.toggleDescriptionHandler }
          >
            <Icon name='cancel' />
          </div>

          <header className={
            classNames(
              styles.header,
              { [styles.headerOffset]: this.state.descriptionOpen },
            )
          }>
            <h1 className={ styles.title }>{ this.props.movie.title }</h1>
            <p className={ styles.genres }>{
              this.props.movie.genres.slice(0, 3).map(genre => (
                <span className={ styles.genre }>
                  { genre.name }
                </span>
              ))
            }</p>
            <div className={ styles.rating }>
              <Rating value={ this.props.movie.vote_average / 2 }/>
            </div>
          </header>

          <p className={
            classNames(
              styles.descriptionText,
              { [styles.descriptionTextVisible]: !this.state.descriptionOpen },
            )
          }>
            { this.props.movie.overview }
          </p>

          <Button
            label='Watch Now'
            className={ styles.watchButton }
            onClick={ () => this.props.openDialog(<Video id={ this.props.movie.id }/>) }
          />
        </div>
      </div>
    );
  }
}

MovieListItem.defaultProps = {
  movie: null,
  genres: [],
};

MovieListItem.propTypes = {
  movie: PropTypes.object,
  genres: PropTypes.array,
};

export default withConditionalRendering(
  withDialogContext(MovieListItem),
  ['movie'],
);
