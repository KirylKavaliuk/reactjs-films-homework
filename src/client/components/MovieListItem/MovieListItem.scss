@import 'styles/palette';

.movie-item {
  position: relative;
  width: 100%;
  min-height: 400px;
  margin: 10px 0;
  color: $white;
}

.poster {
  width: 100%;
  height: 400px;
  border-radius: 3px;
  object-fit: cover;
  object-position: center 30%;
  filter: brightness(0.6);
}

.info {
  position: absolute;
  z-index: 2;
  top: 60px;
  left: 0;
  display: flex;
  flex-direction: column;
  box-sizing: border-box;
  width: 100%;
  padding: 20px;
  transition: all 0.45s ease;
  align-items: flex-start;
  justify-content: flex-start;
}

.title {
  position: absolute;
  margin: 0;
  top: -60px;
  left: 0;
  overflow: hidden;
  box-sizing: border-box;
  width: calc(100% - 10px);
  padding: 20px;
  white-space: nowrap;
  text-transform: uppercase;
  text-overflow: ellipsis;
  font-size: 40px;
  line-height: 1.2;
}

.genres {
  color: $light_blue;
  font-size: 18px;
}

.cancel-icon {
  display: none;
}

.more {
  display: none;
}

.description-text {
  overflow: auto;
  width: 100%;
  max-width: 400px;
  max-height: 150px;
  margin-bottom: 20px;
  text-indent: 20px;
  line-height: 1.35;

  &::-webkit-scrollbar {
    display: none;
  }
}

.genre:hover {
  cursor: pointer;
  color: $lighter_blue;
}

.genre::after {
  content: ', ';
}

.genre:last-child::after {
  content: '';
}
