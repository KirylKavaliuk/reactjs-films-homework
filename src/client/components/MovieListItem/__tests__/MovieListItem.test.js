import React from 'react';
import renderer from 'react-test-renderer';
import { MemoryRouter } from 'react-router-dom';
import MovieListItem from '../MovieListItem';

it('renders movie list item correctly', () => {
  const tree = renderer.create(
    <MemoryRouter>
      <MovieListItem
        movie={{
          id: '1',
          poster_path: '/5Kg76ldv7VxeX9YlcQXiowHgdX6.jpg',
          title: 'title',
          genres: [
            {id: 1, name: 'genre1'},
            {id: 2, name: 'genre2'},
            {id: 3, name: 'genre3'},
          ],
          vote_average: 3.5,
          overview: 'overview',
        }}
      />
    </MemoryRouter>,
  );

  const movieGridItem = tree.toJSON();
  expect(tree).not.toBeNull();
  expect(movieGridItem).toMatchSnapshot();

});

it('renders movie list item uncorrectly', () => {
  const tree = renderer.create(
    <MovieListItem/>,
  ).toJSON();

  expect(tree).toBeNull();
  expect(tree).toMatchSnapshot();
});