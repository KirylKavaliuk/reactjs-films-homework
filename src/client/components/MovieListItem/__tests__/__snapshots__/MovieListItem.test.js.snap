// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MovieListItem/> 1`] = `
<div
  className="MovieListItem__movie-item___3ihj9"
>
  <div
    className="MovieListItem__info___fbjZF"
  >
    <header>
      <h1
        className="MovieListItem__title___2J99z"
      >
        title
      </h1>
      <p
        className="MovieListItem__genres___XBK81"
      >
        <a
          className="Link__link___-IU0S MovieListItem__genre____sUKh"
          href="/genre/1"
          onClick={[Function]}
          onMouseEnter={[Function]}
        >
          genre1
        </a>
        <a
          className="Link__link___-IU0S MovieListItem__genre____sUKh"
          href="/genre/2"
          onClick={[Function]}
          onMouseEnter={[Function]}
        >
          genre2
        </a>
        <a
          className="Link__link___-IU0S MovieListItem__genre____sUKh"
          href="/genre/3"
          onClick={[Function]}
          onMouseEnter={[Function]}
        >
          genre3
        </a>
      </p>
      <div>
        <div
          className="Rating__rating___3f4Qx"
        >
          <div
            className="Rating__star___3Zoo4 Rating__active___BzsSn"
            onClick={[Function]}
            onMouseOut={[Function]}
            onMouseOver={[Function]}
          />
          <div
            className="Rating__star___3Zoo4 Rating__active___BzsSn"
            onClick={[Function]}
            onMouseOut={[Function]}
            onMouseOver={[Function]}
          />
          <div
            className="Rating__star___3Zoo4"
            onClick={[Function]}
            onMouseOut={[Function]}
            onMouseOver={[Function]}
          />
          <div
            className="Rating__star___3Zoo4"
            onClick={[Function]}
            onMouseOut={[Function]}
            onMouseOver={[Function]}
          />
          <div
            className="Rating__star___3Zoo4"
            onClick={[Function]}
            onMouseOut={[Function]}
            onMouseOver={[Function]}
          />
          <div
            className="Rating__rating-number-wrap___dhLPE"
          >
            <div
              className="Rating__rating-number___13U9X"
            >
              1.75
            </div>
          </div>
        </div>
      </div>
    </header>
    <p
      className="MovieListItem__description-text___YTjIk"
    >
      overview
    </p>
    <a
      className="Link__link___-IU0S"
      href="/"
      onClick={[Function]}
      onMouseEnter={[Function]}
    >
      <button
        className="Button__button___1nbn1"
        onClick={null}
      >
        Watch Now
      </button>
    </a>
  </div>
</div>
`;

exports[`<MovieListItem/> 2`] = `null`;
